src/screen.rs

```diff
@@ -1078,10 +1078,30 @@ impl Screen {
         self.need_clear_lines = false;
         self.need_clear_screen = false;
 
+        // Compare the number of lines in the (new) left prompt compared to the
+        // current prompt before it's dropped so that the lines can be cleared
+        // appropriately later (see #11875). Avoid unnecessary clearing by only
+        // doing this when `is_final_rendering`.
+        let mut prompt_has_fewer_lines = is_final_rendering;
```

Comment:
If my other suggestions make sense,
then the variable name is not super honest.
Better make it `final_prompt_has_fewer_lines`.